src/VirtoCommerce.News.Web/App/src/modules/vc-news/pages/news-article-details.vue

```diff
@@ -0,0 +1,260 @@
+  (event: "collapse:blade"): void;
+  (event: "expand:blade"): void;
+  (event: "close:blade"): void;
+}
+
+const emit = defineEmits<Emits>();
+
+export interface Props {
+  expanded?: boolean;
+  closable?: boolean;
+  param?: string;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  expanded: true,
+  closable: true,
+  param: undefined
+});
+
+defineOptions({
+  url: "/vc-news-details",
+  name: "NewsArticleDetails",
+});
+
+const { t } = useI18n({ useScope: "global" });
+
+//stores
+const { stores, loadStores, loadingStores } = useStore();
+const storeOptions = computed(() => stores.value.map((x) => ({ id: x.id, title: x.name })));
+
+//userGroups
+const { userGroups, loadUserGroups, loadingUserGroups } = useUserGroups();
+const userGroupsOptions = computed(() => userGroups.value.map((x) => ({ id: x, title: x })));
+
+const userGroupsSelected = computed({
+  get() {
+    return newsArticle.value?.userGroups?.map((x) => ({ id: x, title: x })) ?? [];
+  },
+  set(newValue) {
+    newsArticle.value!.userGroups = newValue.map((x) => x.id);
+  }
+});
+
+//localization
+const { languages, loadLanguages, loadingLanguages } = useLocalization();
+
+const currentLocale = ref<string>("en-US");
+const setLocale = (locale: string) => {
+  currentLocale.value = locale;
+};
+
+const selectedLocalizedContent = computed(
+  () => {
+    if (newsArticle.value) {
+      if (!newsArticle.value.localizedContents) {
+        newsArticle.value.localizedContents = [];
+      }
+    }
+    const existingLocalizedContent = newsArticle.value?.localizedContents?.find((x) => x.languageCode === currentLocale.value);
+
+    if (existingLocalizedContent) {
+      return existingLocalizedContent;
+    }
+
+    const newLocalizedContent = new NewsArticleLocalizedContent();
+    newLocalizedContent.languageCode = currentLocale.value;
+    newsArticle.value?.localizedContents?.push(newLocalizedContent);
+    return newLocalizedContent;
+  }
+);
+
+const selectedSeo = computed(
+  () => {
+    if (newsArticle.value) {
+      if (!newsArticle.value.seoInfos) {
+        newsArticle.value.seoInfos = [];
+      }
+    }
+    const existingSeoInfo = newsArticle.value?.seoInfos?.find((x) => x.languageCode === currentLocale.value);
+
+    if (existingSeoInfo) {
+      return existingSeoInfo;
+    }
+
+    const newSeoInfo = new SeoInfo();
+    newSeoInfo.isActive = false;
+    newSeoInfo.languageCode = currentLocale.value;
+    newSeoInfo.storeId = newsArticle.value.storeId;
+    newsArticle.value?.seoInfos?.push(newSeoInfo);
+    return newSeoInfo;
+  }
+);
+
+//news article 
+const { newsArticle, loadNewsArticle, saveNewsArticle, loadingOrSavingNewsArticle, publishNewsArticle, unpublishNewsArticle, newsArticleCanPublish, newsArticleCanUnpublish, newsArticleIsDirty, resetNewsArticle } = useNewsArticleDetails();
+
+
+//other
+const loading = computed(() => loadingStores.value || loadingUserGroups.value || loadingLanguages.value || loadingOrSavingNewsArticle.value);
```

Comment:
For convenience, you can use the useLoading composable from @vc-shell/framework here.